torch/csrc/distributed/rpc/init.cpp

```diff
@@ -302,17 +303,21 @@ If the future completes with an error, an exception is thrown.
          const std::shared_ptr<torch::autograd::profiler::RecordFunction>& rf,
          const py::args& args,
          const py::kwargs& kwargs) {
+        DCHECK(!PyGILState_Check());
```

Comment:
Just want to make sure that I understand this correctly. The behavior is if the GIL is held at this point, it will crash immediately?

Reply:
I believe the behaviour is that if the GIL is held by "this thread" then it will crash.

Reply:
This would crash only in "debug" mode. I used DCHECK to avoid any perf overheads in prod. In prod, this check wouldn't even execute.